File: src/Services/Basket/Basket.API/Data/Entities/ShoppingCart.cs
using System.Collections;
using System.ComponentModel.DataAnnotations.Schema;

namespace Basket.API.Data.Entities
{
    public class ShoppingCart
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public List<ShoppingCartItem> Items { get; set; } = new List<ShoppingCartItem>();

        public ShoppingCart(int userId)
        {
            UserId = userId;
        }

        [NotMapped]
        public decimal TotalPrice
        {
            get
            {
                var totalPrice = 0m;
                foreach (var item in Items)
                {
                    totalPrice += item.UnitPrice * item.Quantity;
                }
                return totalPrice;
            }
        }
    }
}

File: src/UI/ShopperHub/src/models/shopping-cart.ts
import { CartItem } from "./cart-item";

export class ShoppingCart {
    constructor(
        userId: number = 0,
        totalPrice: number = 0,
        items: CartItem[] = [],

    ) {
        this.userId = userId;
        this.totalPrice = totalPrice;
        this.items = items;
    }

    userId: number;
    totalPrice: number;
    items: CartItem[];
}